backend/common/src/main/java/ai/verta/modeldb/common/futures/FutureExecutor.java

```diff
@@ -35,19 +41,25 @@ public static FutureExecutor newSingleThreadExecutor() {
 
   @Override
   public void execute(@NonNull Runnable r) {
+    if (otelContext != null) {
+      r = otelContext.wrap(r);
```

Comment:
this is nice. I think this will let us clean up a ton of cruft in the InternalFuture itself that was doing this the hard way.